klib/ntp.c

```diff
@@ -677,17 +724,19 @@ define_closure_function(0, 2, void, ntp_query_func,
         rprintf("NTP: failed to receive server response\n", __func__);
         ntp_query_complete(false);
     }
-    ip_addr_t server_addr;
-    lwip_lock();
-    ntp_server server = vector_get(ntp.servers, ntp.current_server);
-    err_t err = dns_gethostbyname(server->addr, &server_addr, ntp_dns_cb, 0);
-    lwip_unlock();
-    if (err == ERR_OK)
-        ntp_query(&server_addr, false);
-    else if (err != ERR_INPROGRESS) {
-        rprintf("%s: failed to resolve hostname: %d\n", __func__, err);
-        ntp_query_complete(false);
+    int current_server = ntp.current_server;
```

Comment:
I think you need an ntp_lock() around this section for the timer handler.

Reply:
It seems to me a lock would only be needed if we weren't using a local variable (`current_server`) and were using `ntp.current_server` directly. But since the `ntp.current_server` value here is only read once when initializing the local variable (and the `ntp.servers` vector is never modified after init), it seems we can do safely without the lock.

Reply:
That makes sense. Works for me.